#! /bin/sh /usr/share/dpatch/dpatch-run
## sqlite3-support.dpatch by Diego Iastrubni <diego.iastrubni@xorcom.com>
##
## DP: Add SQLite3 support. Fixes upstream http://freepbx.org/trac/ticket/1796

@DPATCH@
diff -urNad freepbx-module-cidlookup~/install.php freepbx-module-cidlookup/install.php
--- freepbx-module-cidlookup~/install.php	2006-06-25 19:04:35.000000000 +0300
+++ freepbx-module-cidlookup/install.php	2007-02-20 11:02:01.000000000 +0200
@@ -1,15 +1,69 @@
 <?php
 
 global $db;
+global $amp_conf;
+
+$autoincrement = (($amp_conf["AMPDBENGINE"] == "sqlite") || ($amp_conf["AMPDBENGINE"] == "sqlite3")) ? "AUTOINCREMENT":"AUTO_INCREMENT";
+
+// create the tables
+$sql = "CREATE TABLE IF NOT EXISTS cidlookup (
+	cidlookup_id INTEGER NOT NULL PRIMARY KEY $autoincrement,
+	description varchar(50) NOT NULL,
+	sourcetype varchar(100) NOT NULL,
+	cache tinyint(1) NOT NULL default '0',
+	deptname varchar(30) default NULL,
+	http_host varchar(30) default NULL,
+	http_port varchar(30) default NULL,
+	http_username varchar(30) default NULL,
+	http_password varchar(30) default NULL,
+	http_path varchar(100) default NULL,
+	http_query varchar(100) default NULL,
+	mysql_host varchar(60) default NULL,
+	mysql_dbname varchar(60) default NULL,
+	mysql_query text,
+	mysql_username varchar(30) default NULL,
+	mysql_password varchar(30) default NULL
+);";
+$check = $db->query($sql);
+if (DB::IsError($check)) {
+        die( "Can not create `cidlookup` table: " . $check->getMessage() .  "\n");
+}
+
+
+$sql = "CREATE TABLE IF NOT EXISTS cidlookup_incoming (
+	cidlookup_id INT NOT NULL,
+	extension VARCHAR(50),
+	cidnum VARCHAR(30),
+	channel VARCHAR(30)
+);";
+$check = $db->query($sql);
+if (DB::IsError($check)) {
+        die( "Can not create `cidlookup_incomming` table: " . $check->getMessage() .  "\n");
+}
 
+// first update
 $sql = "SELECT cache FROM cidlookup";
 $check = $db->getRow($sql, DB_FETCHMODE_ASSOC);
-if(DB::IsError($check)) {
+if (DB::IsError($check)) {
 	// add new field
-	$sql = "ALTER TABLE cidlookup ADD cache TINYINT( 1 ) NOT NULL DEFAULT 0;";
+	$sql = "ALTER TABLE cidlookup ADD cache INTEGER NOT NULL DEFAULT 0;";
 	$result = $db->query($sql);
 	if(DB::IsError($result)) {
-		die($result->getDebugInfo());
+		die($result->getMessage());
 	}
 }
 
+// second update
+$sql = "SELECT cidlookup from incoming;";
+$check = $db->query($sql);
+if (DB::IsError($check)) {
+	$sql = "ALTER TABLE incoming ADD cidlookup INT(2);";
+	$check = $db->query($sql);
+	if (DB::IsError($check)) {
+		die( "Can not alter `incoming` table: " . $check->getMessage() .  "\n");
+	}
+}
+
+
+?>
+
diff -urNad freepbx-module-cidlookup~/install.sql freepbx-module-cidlookup/install.sql
--- freepbx-module-cidlookup~/install.sql	2006-06-25 17:56:59.000000000 +0300
+++ freepbx-module-cidlookup/install.sql	1970-01-01 02:00:00.000000000 +0200
@@ -1,29 +0,0 @@
-ALTER TABLE incoming ADD cidlookup INT(2);
-
-CREATE TABLE IF NOT EXISTS cidlookup (
-  cidlookup_id int(11) NOT NULL auto_increment,
-  description varchar(50) NOT NULL,
-  sourcetype varchar(100) NOT NULL,
-  cache tinyint(1) NOT NULL default '0',
-  deptname varchar(30) default NULL,
-  http_host varchar(30) default NULL,
-  http_port varchar(30) default NULL,
-  http_username varchar(30) default NULL,
-  http_password varchar(30) default NULL,
-  http_path varchar(100) default NULL,
-  http_query varchar(100) default NULL,
-  mysql_host varchar(60) default NULL,
-  mysql_dbname varchar(60) default NULL,
-  mysql_query text,
-  mysql_username varchar(30) default NULL,
-  mysql_password varchar(30) default NULL,
-  PRIMARY KEY  (cidlookup_id)
-);
-
-
-CREATE TABLE IF NOT EXISTS cidlookup_incoming (
-  cidlookup_id INT NOT NULL,
-	extension VARCHAR(50),
-	cidnum VARCHAR(30),
-	channel VARCHAR(30)
-);
diff -urNad freepbx-module-cidlookup~/uninstall.sql freepbx-module-cidlookup/uninstall.sql
--- freepbx-module-cidlookup~/uninstall.sql	2006-06-02 20:04:01.000000000 +0300
+++ freepbx-module-cidlookup/uninstall.sql	2007-02-20 11:01:50.000000000 +0200
@@ -1,4 +1,4 @@
-DROP TABLE cidlookup;
-DROP TABLE cidlookup_incoming;
+DROP TABLE IF EXISTS cidlookup;
+DROP TABLE IF EXISTS cidlookup_incoming;
 
 
